packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts

```diff
@@ -0,0 +1,119 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+import WorkflowSession from '@cardstack/web-client/models/workflow/workflow-session';
+import Layer2Network from '@cardstack/web-client/services/layer2-network';
+import MerchantCustomizationService, {
+  MerchantCustomization,
+} from '@cardstack/web-client/services/merchant-customization';
+import { isLayer2UserRejectionError } from '@cardstack/web-client/utils/is-user-rejection-error';
+import {
+  RegisterMerchantOptions,
+  TransactionHash,
+} from '@cardstack/web-client/utils/web3-strategies/types';
+
+import { task, TaskGenerator } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { reads } from 'macro-decorators';
+
+// http://ember-concurrency.com/docs/typescript
+// infer whether we should treat the return of a yield statement as a promise
+type Resolved<T> = T extends PromiseLike<infer R> ? R : T;
+interface CardPayCreateMerchantWorkflowPrepaidCardChoiceComponentArgs {
+  workflowSession: WorkflowSession;
+  onComplete: () => void;
+  isComplete: boolean;
+}
+
+export default class CardPayCreateMerchantWorkflowPrepaidCardChoiceComponent extends Component<CardPayCreateMerchantWorkflowPrepaidCardChoiceComponentArgs> {
+  @service declare merchantCustomization: MerchantCustomizationService;
+  @service declare layer2Network: Layer2Network;
+
+  @tracked chinInProgressMessage?: string;
+  @tracked txHash?: TransactionHash;
+
+  @reads('createTask.last.error') declare error: Error | undefined;
+
+  @action createMerchant() {
+    taskFor(this.createTask)
+      .perform()
+      .catch((e) => console.error(e));
+  }
+
+  @task *createTask(): TaskGenerator<void> {
+    let { workflowSession } = this.args;
+
+    try {
+      this.chinInProgressMessage =
+        'Preparing to create your custom prepaid card…';
```

Comment:
This message is not correct.

Reply:
hah indeed, updated now to `Preparing to create merchant…`